source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -308,28 +309,32 @@ FilterDataStatus Filter::onData(ProcessorState& state, Buffer::Instance& data, b
       state.setPaused(true);
       result = FilterDataStatus::StopIterationNoBuffer;
     } else if (end_stream || state.queueOverHighLimit()) {
-      switch (openStream()) {
-      case StreamOpenState::Error:
-        return FilterDataStatus::StopIterationNoBuffer;
-      case StreamOpenState::IgnoreError:
-        return FilterDataStatus::Continue;
-      case StreamOpenState::Ok:
-        // Fall through
-        break;
+      bool terminate;
+      std::tie(terminate, result) = sendStreamChunk(state, data, end_stream);
+
+      if (terminate) {
+        return result;
       }
-      state.enqueueStreamingChunk(data, false, false);
-      // Put all buffered data so far into one big buffer
-      const auto& all_data = state.consolidateStreamedChunks(true);
-      ENVOY_LOG(debug, "Sending {} bytes of data in buffered partial mode. end_stream = {}",
-                all_data.data.length(), end_stream);
-      sendBodyChunk(state, all_data.data,
-                    ProcessorState::CallbackState::BufferedPartialBodyCallback, end_stream);
-      result = FilterDataStatus::StopIterationNoBuffer;
-      state.setPaused(true);
     } else {
       // Keep on running and buffering
       state.enqueueStreamingChunk(data, false, false);
-      result = FilterDataStatus::Continue;
+
+      if (Runtime::runtimeFeatureEnabled(Runtime::defer_processing_backedup_streams) &&
+          state.queueOverHighLimit()) {
+        // We just transitioned to queue over high limit, which will then cause
+        // the receiving codec to buffer the data rather than send it to us.
+        // Prior, we'd rely on eager processing to kick off sending the data to
+        // the external processor, instead we kick it off here.
+        bool terminate;
+        Buffer::OwnedImpl empty_buffer{};
```

Comment:
Looks like this code isn't covered (https://storage.googleapis.com/envoy-pr/a8ec184/coverage/source/extensions/filters/http/ext_proc/ext_proc.cc.gcov.html) by tests.

Reply:
Good eyes! was missing parameterize the streaming test which uses this